client/src/components/MainPage/TaskModal/TagInput.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+import { HOST } from '../../../constants';
+import styled from 'styled-components';
+import useInput from '../../../hooks/useInput';
+import { RiCloseLine } from 'react-icons/ri';
+import { Tag } from 'GlobalType';
+
+axios.defaults.withCredentials = true; // withCredentials 전역 설정
+
+interface Props {
+  setTagIdx: React.Dispatch<React.SetStateAction<number | undefined>>;
+}
+
+const TagInput = (props: Props) => {
+  const [tagList, setTagList] = useState<Tag[]>([]);
+  const [tagInput, onTagInputChange, setTagInput] = useInput('');
+  const [isTagFocus, setIsTagFocus] = useState<Boolean>(false); // Input창이 Focus될 때만 List 표시
```

Comment:
변수명이 길어지더라도 의미하는 바가 명확하면 좋을 것 같습니다.
```suggestion
  const [isTagInputFocused, setIsTagInputFocused] = useState<Boolean>(false); // Input창이 Focus될 때만 List 표시
```

Reply:
변수명, null 이슈 반영했습니다 - https://github.com/boostcampwm-2022/web19-Boostart/pull/57/commits/806da3b5a31969e220a3e3c274f4eec3d3c24272